### PR TITLE
Use assertThatThrownBy in TestQueryStateInfoResource

### PR DESCRIPTION
**Pull request by Damans227**
Linked to `@Test(expectedExceptions...)` cleanup issue #6974

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/server/TestQueryStateInfoResource.java [hunk 1]
```diff
@@ -225,14 +225,11 @@ public void testGetQueryStateInfoDenied()
...
+    @Test
     public void testGetQueryStateInfoNo()
     {
-        client.execute(
-                prepareGet()
-                        .setUri(server.resolve("/v1/queryState/123"))
-                        .setHeader(TRINO_HEADERS.requestUser(), "unknown")
-                        .build(),
-                createJsonResponseHandler(jsonCodec(QueryStateInfo.class)));
+        assertThatThrownBy(() -> client.execute(prepareGet().setUri(server.resolve("/v1/queryState/123")).setHeader(TRINO_HEADERS.requestUser(), "unknown").build(), createJsonResponseHandler(jsonCodec(QueryStateInfo.class))))
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/server/TestQueryStateInfoResource.java:231]: This line is too long. Let's format like below (same as the above `testGetQueryStateInfoDenied()` method)
```java
        assertThatThrownBy(() -> client.execute(
                prepareGet()
                        .setUri(server.resolve("/v1/queryState/123"))
                        .setHeader(TRINO_HEADERS.requestUser(), "unknown")
                        .build(),
                createJsonResponseHandler(jsonCodec(QueryStateInfo.class))))
                .isInstanceOf(UnexpectedResponseException.class)
                .hasMessageMatching("Expected response code .*, but was 404");
```

**Review by ebyhr — APPROVED** [review]
Looks good except for a minor comment.

**Review by Damans227 — COMMENTED** [review]
Makes sense, thanks for the review.

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**